### PR TITLE
Add inverse trig functions to logic

### PR DESCRIPTION
**Pull request by nilederg**

### CODE REVIEW

File: core/src/mindustry/logic/LogicOp.java [hunk 1]
```diff
@@ -37,8 +37,11 @@
     log("log", Math::log),
     log10("log10", Math::log10),
     sin("sin", d -> Math.sin(d * 0.017453292519943295D)),
+    arcsin("arcsin", d -> Math.asin(d) / 0.017453292519943295D),
     cos("cos", d -> Math.cos(d * 0.017453292519943295D)),
+    arccos("arccos", d -> Math.acos(d) / 0.017453292519943295D),
     tan("tan", d -> Math.tan(d * 0.017453292519943295D)),
+    arctan("arctan", d -> Mathf.atan(d) / 0.017453292519943295D),
```
Comment (Quezler) [inline core/src/mindustry/logic/LogicOp.java:44]: maybe its in `Math` instead of `Mathf` ?

File: core/src/mindustry/logic/LogicOp.java [hunk 2]
```diff
@@ -37,8 +37,11 @@
     log("log", Math::log),
     log10("log10", Math::log10),
     sin("sin", d -> Math.sin(d * 0.017453292519943295D)),
+    arcsin("arcsin", d -> Math.asin(d) / 0.017453292519943295D),
```
Comment (summetdev) [inline core/src/mindustry/logic/LogicOp.java:40]: Maybe use `Math.PI / 180` instead of 'magic' numbers?

**Comment by Anuken** [comment]
[Read the contribution guidelines - specifically, about testing your changes.](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md#always-test-your-changes)

What is the use case for these functions in logic? I've written several games and I've never had to use any of them.

**Comment by nilederg** [comment]
I am actually using these, in my case I'm using them for a 3d renderer.

**Comment by nilederg** [comment]
Also, I apologize for not testing first, this is actually my first time using github.

**Comment by Anuken** [comment]
Implemented separately, with shorter names & documentation.

**Pull request closed without merging**